File: be_camphub/src/main/java/org/camphub/be_camphub/mapper/ItemLogMapper.java
package org.camphub.be_camphub.mapper;

import org.camphub.be_camphub.dto.response.item.ItemLogsResponse;
import org.camphub.be_camphub.entity.ItemLog;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper(
        componentModel = "spring",
        uses = {Utils_Mapper.class, MediaResourceMapper.class})
public interface ItemLogMapper {
    @Mapping(target = "account", ignore = true) // sẽ set trong service
    @Mapping(target = "itemName", ignore = true) // sẽ set trong service
    @Mapping(target = "action", source = "action", qualifiedByName = "enumToString")
    @Mapping(target = "previousStatus", source = "previousStatus", qualifiedByName = "enumToString")
    @Mapping(target = "currentStatus", source = "currentStatus", qualifiedByName = "enumToString")
    @Mapping(target = "createdAt", source = "createdAt", qualifiedByName = "localDateTimeToString")
    @Mapping(target = "media", source = "evidenceUrls", qualifiedByName = "toResponse")
    ItemLogsResponse entityToResponse(ItemLog itemLog);
}

File: fe_camphub/src/libs/core/mapping/item-log.mapper.ts
import { createMap, createMapper, forMember, mapFrom } from "@automapper/core";
import { pojos, PojosMetadataMap } from "@automapper/pojos";
import { ItemLog } from "../types";
import { ItemLogsResponse } from "../dto/response";

const itemLogMapper = createMapper({
    strategyInitializer: pojos(),
});

PojosMetadataMap.create<ItemLog>("ItemLog", {
    id: String,
    itemId: String,
    itemName: String,
    account: String,
    action: String,
    previousStatus: String,
    currentStatus: String,
    note: String,
    media: Array,
    createdAt: String,
});
PojosMetadataMap.create<ItemLogsResponse>("ItemLogsResponse", {
    id: String,
    itemId: String,
    itemName: String,
    account: String,
    action: String,
    previousStatus: String,
    currentStatus: String,
    note: String,
    media: Array,
    createdAt: String,
});

createMap<ItemLogsResponse, ItemLog>(
    itemLogMapper,
    "ItemLogsResponse",
    "ItemLog",
);

export const itemLogMap = {
    fromResponse: (response: ItemLogsResponse): ItemLog => {
        return itemLogMapper.map(response, "ItemLogsResponse", "ItemLog");
    }
}